src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package com.synacy.lesson02.exercises;
+
+import com.synacy.lesson02.exercises.domain.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.xml.ws.soap.MTOM;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+
+    private EnrollmentService enrollmentService;
+    private EnrollmentDto enrollmentDetails;
+
+    @Mock StudentProfileService studentProfileService;
+    @Mock CourseClassService courseClassService;
+    @Mock EnrollmentNotificationService enrollmentNotificationService;
+    @Mock StudyLoadFormatter studyLoadFormatter;
+    @Mock SystemService systemService;
+
+    @Before
+    public void setup(){
+        enrollmentService = new EnrollmentService();
+
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+        enrollmentService.setSystemService(systemService);
+        enrollmentDetails = mock(EnrollmentDto.class);
+
+    }
+
+    @Test
+    public void processEnrollment_shouldUpdateStudentProfileWithEnrollmentDetails() throws Exception {
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+        verify(studentProfileService).updateStudentProfile(enrollmentDetails.getStudentProfile());
+    }
+
+    @Test
+    public void processEnrollment_shouldEnrollStudentClasses() throws Exception {
+
+        Student student = mock(Student.class);
+
+        Set<CourseClass> courseSet = new HashSet<>();
+        courseSet.add(mock(CourseClass.class));
+        Iterator<CourseClass> iterator = courseSet.iterator();
+
+        when(enrollmentDetails.getEnrolledClasses()).thenReturn(courseSet);
+        when(enrollmentDetails.getStudent()).thenReturn(student);
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        while (iterator.hasNext()){
+            verify(courseClassService).enrollStudentToClass(
+                    (enrollmentDetails.getStudent()),
+                    (iterator.next())
+            );
+        }
+    }
```

Comment:
A better test implementation is to get rid of the `Iterator` such as one shown below. Using the Iterator interface provides additional dependency for this test case method (this test knows another class) which we would rather minimize.

![much better](https://cloud.githubusercontent.com/assets/7790345/25770776/fe4fbcbe-3270-11e7-81f6-7c27304612da.png)